src/panels/lovelace/cards/hui-clock-card.ts

```diff
@@ -0,0 +1,256 @@
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import "../../../components/ha-alert";
+import "../../../components/ha-card";
+import type { HomeAssistant } from "../../../types";
+import type {
+  LovelaceCard,
+  LovelaceCardEditor,
+  LovelaceGridOptions,
+} from "../types";
+import type { ClockCardConfig } from "./types";
+import { TimeFormat } from "../../../data/translation";
+import { useAmPm } from "../../../common/datetime/use_am_pm";
+import { resolveTimeZone } from "../../../common/datetime/resolve-time-zone";
+
+const INTERVAL = 1000;
+
+@customElement("hui-clock-card")
+export class HuiClockCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import("../editor/config-elements/hui-clock-card-editor");
+    return document.createElement("hui-clock-card-editor");
+  }
+
+  public static getStubConfig(): ClockCardConfig {
+    return {
+      type: "clock",
+      clock_size: "medium",
+      show_seconds: true,
+      time_format: TimeFormat.am_pm,
```

Comment:
Lets not put this in the stub config, keep it as small as possible, just `type: "clock"` is enough